Tracking: add a should_enable_tracking() method

kbrown9 opened the pull request:
#### Changes proposed in this Pull Request:
* Add a new method, `Tracking::should_enable_tracking()`, which replicates the original logic in the `Terms_Of_Service::has_agreed()` method [before the changes](https://github.com/Automattic/jetpack/pull/16967/files#diff-70716275c6e734611a8a24650646c4fbL62) in PR #16967.
* Add unit tests for the new `Tracking::should_enable_tracking()` method.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* No.

#### Testing instructions:
1. Install and activate Jetpack.
2. In your `wp-config.php` file, set the `JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME` constant to true. This will force the Calypso connection process to be used.
3. On the Jetpack dashboard, click the `Set up Jetpack` button.
4. Go to the Tracks live view and verify that the `jetpack_jpc_register_success` event was recorded.

#### Proposed changelog entry for your changes:
* tbd

zinigor commented:
I tested the code just in case we might need a point release, and it's working well for me. The tracking events get sent even when the connection is not established yet, which is the expected behavior. Thanks! Are you planning to make more changes, or can we flag it for review?

kbrown9 commented:
> I tested the code just in case we might need a point release, and it's working well for me. The tracking events get sent even when the connection is not established yet, which is the expected behavior. Thanks! Are you planning to make more changes, or can we flag it for review?

Thanks for taking a look at this! I added test instructions and the `Needs Review` label. I also added this to the 8.9.1 milestone in case we release 8.9.1.

sergeymitr reviewed (approved):
Looks great! 🚢

jeherve commented on packages/tracking/composer.json:
```diff
@@ -5,6 +5,7 @@
 	"require": {
 		"automattic/jetpack-options": "@dev",
+		"automattic/jetpack-status": "@dev",
 		"automattic/jetpack-terms-of-service": "@dev"
```
Since we're not using `Terms_Of_Service` in the package anymore, should we get rid of that dependency?

zinigor replied:
We do use it when creating the instance to pass into the new method [here](https://github.com/Automattic/jetpack/blob/master/packages/tracking/src/class-tracking.php#L103). Thanks for being mindful of the dependencies, it's good to have a sanity check on those every once in a while!

jeherve commented on packages/tracking/src/class-tracking.php:
```diff
@@ -103,11 +101,9 @@ public function tracks_record_event( $user, $event_name, $properties = array(),
-			! $terms_of_service->has_agreed()
-			|| ! $this->connection->is_user_connected()
-		) {
+		$status           = new Status();
```
Since we're now using `Status()`, should we import it at the top of the file?

zinigor replied:
We don't need to do that because `Status` is in the same namespace as `Tracking` - all classes are assumed to be `\Automattic\Jetpack`.

jeherve reviewed (commented):
I'm late to the party with this, but I had 2 questions before I update the Tracking package and all packages that depend on it.

jeherve commented:
Cherry-picked to `branch-8.9` in e9bf67c35411fdef0fa5a5b3cf660da6d6e350ec

jeherve commented:
I updated the Tracking package version in 94b0e94bda8a278dca9c482b51ef9aecc1615e86, as well as the other packages that depend on Tracking (connection, abtest, jitm, sync).
